src/daemons/rhsmcertd.c

```diff
@@ -504,6 +555,31 @@ key_file_init_config (Config * config, GKeyFile * key_file)
             DEFAULT_AUTO_REGISTRATION
             );
     config->auto_registration = auto_registration_enabled;
+
+    gchar *default_log_level = get_string_from_config_file(
+            key_file,
+            "logging",
+            "default_log_level"
+            );
+
+    gchar *rhsmcertd_log_level = get_string_from_config_file(
+            key_file,
+            "logging",
+            "rhsmcertd"
+            );
+
+    if (rhsmcertd_log_level == NULL && default_log_level != NULL) {
+        set_log_level(default_log_level, "logging.default_log_level");
+    } else if (rhsmcertd_log_level != NULL) {
+        set_log_level(rhsmcertd_log_level, "logging.rhsmcertd");
+    }
```

Comment:
you can avoid one extra check if you reorder the conditions:
```c
    if (rhsmcertd_log_level != NULL) { ... }
    else if (default_log_level != NULL) { ... }
```
which also makes it easier to understand the priority of the keys (`rhsmcertd` first, then `default_log_level`)